File: python/fast-api/src/dice/dice.py
import secrets


class Dice:
    def __init__(self, sides: int) -> None:
        self.sides = sides

    def roll(self) -> int:
        return secrets.randbelow(self.sides) + 1

    def __repr__(self) -> str:
        return f"Dice(sides={self.sides})"

File: rust/axum/src/dice.rs
use rand::Rng;

pub struct Dice {
    pub sides: u32,
}

impl Dice {
    pub fn new(sides: u32) -> Self {
        Self { sides }
    }

    pub fn roll(&self) -> u32 {
        let mut rng = rand::rng();
        rng.random_range(1..=self.sides)
    }
}
